.github/workflows/build-auto-qttest-on-pullrequest.yml

```diff
@@ -97,3 +97,100 @@ jobs:
           echo "-- test Seamly2D with xvfb --"
           export LD_LIBRARY_PATH=/home/runner/local/lib64/Seamly2D
           xvfb-run -a make --silent check TESTARGS="-silent"
+
+  build-windows:
+    # run job 'build-test' for open pull_request that hasn't been merged yet
+    if: ( github.event.pull_request.merged == false )
+    name: 'Windows: Build'
+    runs-on: windows-latest
+
+    steps:
+
+      - name: checkout code
+        uses: actions/checkout@v3
+
+      - name: install msvc
+        uses: ilammy/msvc-dev-cmd@v1.10.0
+        with:
+          version: ${{ env.QT_VERSION }}
+          # use SVC 2019 latest, 14.29.xxxx
+          toolset: '14.29'
+
+      - name: install qt
+        uses: jurplel/install-qt-action@v2.14.0
+        with:
+          version: ${{ env.QT_VERSION }}
+
+      - name: make seamly2d.exe and seamlyme.exe
+        run: |
+          qmake.exe Seamly2D.pro -r CONFIG+=no_ccache CONFIG+=noDebugSymbols
+          nmake
+
+      - name: create seamly2d-installer.exe
+        # seamly2d.exe, seamlyme.exe, VC_redist, OpenSSL dlls, and create + code sign installer
+        # uses parameters in seamly2d-installer.nsi to create the installer exe
+        run: |
+          mkdir ..\windows-build
+          Get-ChildItem -Recurse -Include *.exe,*.dll | % { Copy-Item $_.FullName -force -destination ..\windows-build }
+          windeployqt.exe --libdir ..\windows-build --plugindir ..\windows-build --release ..\windows-build\seamly2d.exe
+          windeployqt.exe --force --libdir ..\windows-build --plugindir ..\windows-build --release ..\windows-build\seamlyme.exe
+          copy .\dist\seamly2d-installer.nsi ..\windows-build\
+          copy .\dist\win\VC_redist.x86.exe ..\windows-build\
+          copy .\dist\win\VC_redist.x64.exe ..\windows-build\
+          copy 'c:\Program Files\OpenSSL\libcrypto-1_1-x64.dll' ..\windows-build\
+          copy 'c:\Program Files\OpenSSL\libssl-1_1-x64.dll' ..\windows-build\
+          copy 'c:\Program Files\OpenSSL\bin\capi.dll' ..\windows-build\
+          copy 'c:\Program Files\OpenSSL\bin\dasync.dll' ..\windows-build\
+          New-Item '..\windows-build\translations' -ItemType Directory -Force | Out-Null
+          Copy-Item 'share\translations\*.qm' '..\windows-build\translations' -Force -Recurse
+          cd ..\windows-build\
+          & 'C:\Program Files (x86)\NSIS\makensis.exe' seamly2d-installer.nsi
+
+      - name: pack installer file into .zip
+        run: |
+          cd ..\windows-build\
+          C:\"Program Files"\7-Zip\7z.exe a Seamly2D-windows.zip Seamly2D-installer.exe
+    
+      - name: Upload Seamly2D-installer.exe artifact
```

Comment:
I really appreciate what you're doing. Please accept this review as a hopeful, supportive, and constructive conversation.

Q: Why build & upload artifacts prior to the PR merge?
Q: Why test the Qt/C++ code on multiple platforms?

Reply:
Q: Why build & upload artifacts prior to the PR merge?

My idea is that I want to be able to see if all artifacts are still working before merging a PR instead of after merging a PR. Then failures like #752 would be avoided, as its visible before the merge whether all platform still build. Why also upload? Then I can easier test refactorings on windows and mac, as I can have the CI built the PR result before merge and just download and try it in a VM. Otherwise I need to have build envs for all platforms locally…
I hope this makes now a bit more sense :)

Q: Why test the Qt/C++ code on multiple platforms?

This may be a misunderstanding, the unit tests only run on linux, not all platforms. I agree that running them on all platforms is not required

Reply:
On the PRs before merge we also dont spam the release history when experimenting

Reply:
Build results can then be downloaded in the overview of the action, eg https://github.com/FashionFreedom/Seamly2D/actions/runs/3816543462

And there I notice that the windows upload still fails, so I will fix this anyway, setting this to draft

Reply:
Now its working fine: https://github.com/FashionFreedom/Seamly2D/actions/runs/3837603378